### PR TITLE
Update elm ui explorer to 8.0.0

### PR DESCRIPTION
**Pull request by jouderianjr**
- Upgrade to 8.0.0
- Improve icons stories
- Add code/note text tab
- Add categories to the stories menu
- Enable/Disable menu if only has one story
- Custom header

### CODE REVIEW

**Comment by jouderianjr** [comment]
I've deployed to our site if you guys want to take a look :D [click here](https://paackeng.github.io/paack-ui)

File: showcase/src/Utils.elm [hunk 1]
```diff
@@ -40,3 +41,8 @@ iconsSvgSprite =
             []
             []
         )
+
+
+prettifyElmCode : String -> String
```
Comment (PedroHLC) [inline showcase/src/Utils.elm:46]: I'm not sure this was the best call, as the notes aren't a simple string but Markdown :smile:

File: showcase/src/Texts.elm [hunk 2]
```diff
@@ -28,13 +32,15 @@ Who will be there to take my place?
 \"\"\"
     |> Text.body1
     |> Text.renderElement renderConfig
-```
-
+"""
+                , note = """
 Futures discussions includes:
```
Comment (PedroHLC) [inline showcase/src/Texts.elm:37]: Please, remove this outdated discussions. :smile:

File: showcase/src/Sizes.elm [hunk 3]
```diff
@@ -4,33 +4,42 @@ import Element
...
-          , { note = notes "small" }
+          , { defaultWithMenu
+                | note = notes "small"
+            }
           )
         , ( "ExtraSmall"
           , \_ -> sizeView cfg Size.extraSmall
-          , { note = notes "extraSmall" }
+          , { defaultWithMenu
+                | note = notes "extraSmall"
```
Comment (PedroHLC) [inline showcase/src/Sizes.elm:41]: Actually, these notes are code samples/tutorial.

File: showcase/src/Palette.elm [hunk 4]
```diff
@@ -13,7 +14,7 @@ stories =
         "Colors"
         [ ( "Colors"
           , \_ -> Element.layout [] colorsView
-          , { note = "" }
+          , defaultWithoutMenu
```
Comment (PedroHLC) [inline showcase/src/Palette.elm:17]: :thinking: 
Can we add code and notes here?

File: showcase/src/Icons.elm [hunk 5]
```diff
@@ -4,63 +4,66 @@ import Element
...
-```elm
+          , { defaultWithoutMenu
+                | code =
+                    prettifyElmCode
+                        """
 Icon.seeMore label
     |> Icon.withColor (Palette.color Palette.tonePrimary Palette.brightnessMiddle)
     |> Icon.renderElement cfg
-```
 """
```
Comment (PedroHLC) [inline showcase/src/Icons.elm:26]: And as notes we could add something like:
```"""
We name icons by their functionality and not their shapes.
```"""

File: showcase/src/Alerts.elm [hunk 6]
```diff
@@ -23,11 +24,10 @@ primaryStory renderConfig =
         ( "Primary"
         , Alert.primary "Hey I just met you"
             |> baseView renderConfig
-        , { note = """
-```elm
+        , { defaultWithMenu
+            | note = prettifyElmCode """
```
Comment (PedroHLC) [inline showcase/src/Alerts.elm:28]: This is not a note, it's code, the other variations are right!

File: showcase/src/LoadingView.elm [hunk 7]
```diff
@@ -50,8 +47,10 @@ largeStory =
         ( "Large"
         , viewBase
             UI.LoadingView.large
-        , { note =
-                """
+        , { defaultWithMenu
+            | code =
+                prettifyElmCode
+                    """
 ```elm
```
Comment (PedroHLC) [inline showcase/src/LoadingView.elm:54]: You forgot to remove the code quoting here.

File: showcase/src/Checkboxes/Stories.elm [hunk 8]
```diff
@@ -73,10 +74,11 @@ boxDemo renderConfig =
                 , ifThenElse box3 0.75 0
                 ]
     in
-    storyWithModel ( "Checkboxes", view, { note = """
-This demo can be reproduced with the following code:
-
-```elm
+    storyWithModel
```
Comment (PedroHLC) [inline showcase/src/Checkboxes/Stories.elm:77]: This storycase made me hungry. :hamburger:

File: showcase/src/Paginators/Stories.elm [hunk 9]
```diff
@@ -65,7 +62,8 @@ Paginator.nonNumeric
     , first = defaultPageSize
     }
     cfg
```
Comment (PedroHLC) [inline showcase/src/Paginators/Stories.elm:64]: better change `cfg` to `renderConfig`

**Comment by PedroHLC** [comment]
Please don't kill me!
?Can we disable the dark-mode button

File: showcase/src/Sizes.elm (hunk 3 above)
Comment (jouderianjr) [reply showcase/src/Sizes.elm:41]: I think this is more a "pseudocode", is not a proper code as it has the comments there. So maybe it makes sense keep it as notes and/or do a complete snippet to put in the code tab.

WDYT?

**Comment by jouderianjr** [comment]
@PedroHLC [click here ㊙️ ](https://github.com/kalutheo/elm-ui-explorer/issues/53)

File: showcase/src/Checkboxes/Stories.elm [hunk 10]
```diff
@@ -49,7 +49,7 @@ boxDemo renderConfig =
                 [ iconsSvgSprite
                 , Checkbox.checkbox renderConfig
                     (Checkboxes.Box1Set >> Msg.CheckboxesStoriesMsg)
-                    "Extra ketchup (+0.50 USD)"
+                    "Extra ketchup (Free)"
```
Comment (PedroHLC) [inline showcase/src/Checkboxes/Stories.elm:52]: :sweat_smile:  thanks

File: showcase/src/Sizes.elm (hunk 3 above)
Comment (PedroHLC) [reply showcase/src/Sizes.elm:41]: If it has code or code with instructions, it is code.
If it has usage notes, warnings, design decisions, it is a note.

File: showcase/src/Utils.elm (hunk 1 above)
Comment (jouderianjr) [reply showcase/src/Utils.elm:46]: the plan here is to use only for `PluginOption.code`

**Comment by PedroHLC** [comment]
I would only do one more change: Add the link to docs on all element's notes...

**Comment by jouderianjr** [comment]
> I would only do one more change: Add the link to docs on all element's notes...

Done, please review again

**Pull request merged**